darwin/DarwinProcess.c

```diff
@@ -327,6 +320,25 @@ void DarwinProcess_setFromKInfoProc(Process* proc, const struct kinfo_proc* ps,
       }
    }
 
+   if (proc->tty_name == NULL && (dev_t)proc->tty_nr != NODEV) {
+      /* The call to devname() is extremely expensive (due to lstat)
+       * and represents ~95% of htop's CPU usage when there is high
+       * process turnover.
+       *
+       * To mitigate this we only fetch TTY information if the TTY
+       * field is enabled in the settings.
+       */
+      if (proc->settings->ss->flags & PROCESS_FLAG_TTY) {
+         const char *name = devname(proc->tty_nr, S_IFCHR);
```

Comment:
Given macos seems to offer `devname_r` I'd prefer to use it. This also avoids that magic number below with `xStrndup` …

Reply:
The magic number (512) is just a pedantic guard and can be removed (it should actually be `sizeof("/dev/") + MAXNAMLEN`) and can be removed.

Using `devname` is nice because we don't need thread-safety and not using it requires duplicating it's logic. The below implementation is essentially the same as in [`devname.c`](https://github.com/apple-oss-distributions/Libc/blob/Libc-1439.141.1/gen/devname.c#L103-L114) itself (note: `devename_r` returns NULL if there is no device hence the fixed sized buffer - otherwise we wouldn't be able to distinguish if NULL is returned because the buffer is too small or there is no device associated with `dev`):

```c
#include <stdlib.h>        // devname_r
#include <string.h>        // strcpy
#include <sys/dirent.h> // MAXNAMLEN

char *get_devname(dev_t dev) {
	char buf[sizeof("/dev/") + MAXNAMLEN];
	char *name = devname_r(dev, S_IFCHR, buf, MAXNAMLEN);
	if (name) {
		return strdup(buf);
	}
	return NULL;
}
```
That said, I don't have a strong opinion here.

Reply:
I think providing our own buffer to `devname_r` instead of copying from some hidden static buffer is preferable. And while we don't strictly need to be re-entrant it doesn't hurt and is basically in line what our code does already anyway.

Reply:
Done